charmcraft/metafiles/metadata.py

```diff
@@ -93,11 +93,13 @@ def create_metadata_yaml(
 
     :returns: Path to created metadata.yaml.
     """
-    file_path = charm_dir / METADATA_FILENAME
-    # metadata.yaml should be copied if it exists in the project
-    if charmcraft_config.metadata_legacy:
+    original_file_path = charmcraft_config.project.dirpath / METADATA_FILENAME
+    target_file_path = charm_dir / METADATA_FILENAME
+
+    # Copy metadata.yaml if it exists, otherwise create it from CharmcraftConfig.
+    if original_file_path.exists():
         try:
-            shutil.copyfile(charmcraft_config.project.dirpath / METADATA_FILENAME, file_path)
+            shutil.copyfile(original_file_path, target_file_path)
         except shutil.SameFileError:
```

Comment:
This could be expressed better with

```
with contextlib.suppress(shutil.SameFileError):
    shutil.copyfile(original_file_path, target_file_path)
```

regardless, I would add a code comment on when this will happen and why it is a good idea to suppress/pass it